axelor-crm/src/main/java/com/axelor/apps/crm/web/LeadController.java

```diff
@@ -190,4 +190,49 @@ public void loseLead(ActionRequest request, ActionResponse response) {
       TraceBackService.trace(response, e);
     }
   }
+
+  public void startLead(ActionRequest request, ActionResponse response) {
+    try {
+      Lead lead = request.getContext().asType(Lead.class);
+      Beans.get(LeadService.class).startLead(Beans.get(LeadRepository.class).find(lead.getId()));
+      response.setReload(true);
+    } catch (Exception e) {
+      TraceBackService.trace(response, e);
+    }
+  }
+
+  public void recycleLead(ActionRequest request, ActionResponse response) {
+    try {
+      Lead lead = request.getContext().asType(Lead.class);
+      Beans.get(LeadService.class).recycleLead(Beans.get(LeadRepository.class).find(lead.getId()));
+      response.setReload(true);
+    } catch (Exception e) {
+      TraceBackService.trace(response, e);
+    }
+  }
+
+  public void assignToMeLead(ActionRequest request, ActionResponse response) {
+    try {
+      Lead lead = request.getContext().asType(Lead.class);
+      Beans.get(LeadService.class)
+          .assignToMeLead(Beans.get(LeadRepository.class).find(lead.getId()));
+      response.setReload(true);
+    } catch (Exception e) {
+      TraceBackService.trace(response, e);
+    }
+  }
+
+  public void assignToMeMultipleLead(ActionRequest request, ActionResponse response) {
+    try {
+      LeadRepository leadRepo = Beans.get(LeadRepository.class);
+      for (Lead lead :
+          leadRepo.all().filter("id in ?1", request.getContext().get("_ids")).fetch()) {
+        Beans.get(LeadService.class)
+            .assignToMeLead(Beans.get(LeadRepository.class).find(lead.getId()));
```

Comment:
Sorry I didn't noticed this earlier. When calling `assignToMeLead` on multiple lead, the process will stop at the first exception and then the next leads will not be treated, but the previous ones would be correctly assigned.

Is it the intended behaviour ?

Reply:
I didn't think about it when I wrote it, but would it be correct to keep it that way ? Or we should cancel the correctly assigned leads when an exception occurs ?

Reply:
We have the same behaviour on fixed asset on mass validation so it is correct to keep it that way, I just wanted to be sure that this was the intended behaviour.

Reply:
Ok, I think I will keep it that way then.